bigcache.go

```diff
@@ -33,6 +33,110 @@ type cacheShard struct {
 	onRemove    func(wrappedEntry []byte)
 }
 
+// EntryInfo holds informations about entry in the cache
+type EntryInfo struct {
+	entry []byte
+}
+
+// Key returns entry's underlying key
+func (e EntryInfo) Key() string {
+	return readKeyFromEntry(e.entry)
+}
+
+// Hash returns entry's hash value
+func (e EntryInfo) Hash() uint64 {
+	return readHashFromEntry(e.entry)
+}
+
+// Timestamp returns entry's timestamp (time of insertion)
+func (e EntryInfo) Timestamp() uint64 {
+	return readTimestampFromEntry(e.entry)
+}
+
+// Value returns entry's underlying value
+func (e EntryInfo) Value() []byte {
+	return readEntry(e.entry)
+}
+
+// EntryInfoIterator allows to iterate over entries in the cache
+type EntryInfoIterator struct {
+	cache        *BigCache
+	currentShard int
+	currentIndex int
+	elements     []uint32
+	sync.Mutex
+}
+
+func copyCurrentShardMap(shard *cacheShard) []uint32 {
+	shard.lock.RLock()
+	defer shard.lock.RUnlock()
+
+	var elements []uint32
+
+	for _, index := range shard.hashmap {
+		elements = append(elements, index)
+	}
+
+	return elements
+}
+
+// Next returns true if there is next element in the iterator
```

Comment:
This describes only half of the job that `Next() ` do becouse it also moves iterator to next element or invalid position if it doesn't exist. This comment better suits `HasNext() `